### DISCUSSION TITLE
How to use optional dependencies like awswrangler[sqlserver] on lambda functions

### DISCUSSION THREAD

**Discussion by AdrianoNicolucci** [aws/aws-sdk-pandas#2631] [Q&A]
AWS SDK for pandas is available as an AWS Lambda Managed layer in all AWS commercial regions. However, it appears it does not contain optional dependencies such as " awswrangler[sqlserver]". What is the recommended way to leverage these optional dependencies on a lambda function? If I attempt to create a custom layer, I don't think it comes with the optional libraries I need which does not solve the problem. 
 
Is there any harm to have these managed layers have them come installed with all optional libraries part of aws SDK for pandas. This would be helpful to many users who interact with rds databases with their lambda functions

**Comment by kukushking** [comment]
Hi @AdrianoNicolucci lambda layers unfortunately have size constraints - 250MB total (unzipped) thus we tend to package only certain most used dependencies with the layers. 

You can create custom layers ([example](https://docs.aws.amazon.com/lambda/latest/dg/python-layers.html)) and specify to include the optional dependencies:
- When installing with pip: `pip install 'awswrangler[sqlserver]==3.11.0'`
- Or in `requirements.txt`: `awswrangler==3.11.0 [sqlserver]`
